apps/web/src/views/AddLiquidityV3/components/AprCalculator.tsx

```diff
@@ -158,13 +158,24 @@ export function AprCalculator({
   const validAmountA = amountA || (inverted ? tokenAmount1 : tokenAmount0) || (inverted ? aprAmountB : aprAmountA)
   const validAmountB = amountB || (inverted ? tokenAmount0 : tokenAmount1) || (inverted ? aprAmountA : aprAmountB)
   const [amount0, amount1] = inverted ? [validAmountB, validAmountA] : [validAmountA, validAmountB]
-  const inRange = isPoolTickInRange(pool, tickLower, tickUpper)
+  const inRange = isPoolTickInRange(pool ?? undefined, tickLower ?? 0, tickUpper ?? 0)
+  const allTicks = useMemo(
+    () =>
+      data?.map(
+        ({ tick, liquidityNet }) => new Tick({ index: parseInt(tick), liquidityNet, liquidityGross: liquidityNet }),
+      ),
+    [data],
+  )
+  const mostActiveLiquidity = useMemo(
+    () => allTicks && sqrtRatioX96 && FeeCalculator.getLiquidityFromSqrtRatioX96(allTicks, sqrtRatioX96),
+    [allTicks, sqrtRatioX96],
+  )
   const { apr } = useRoi({
     tickLower,
     tickUpper,
     sqrtRatioX96,
     fee: feeAmount,
-    mostActiveLiquidity: pool?.liquidity,
+    mostActiveLiquidity,
```

Comment:
hmm the calculated value should match with `pool.liquidity` if the data is correct. I'm thinking probably the ticks data is not up-to-date. What about we switch to use pool liquidity inside the roi calculator?

Reply:
use `pool?.liquidity` is better, on my test it near the same inside and outside